### ISSUE TITLE
Trouble accessing TEXTFIELD from BUTTON code

### ISSUE THREAD

**Issue by jhayes-apl** [OpenC3/cosmos#157]
I have created the following screen.  Note the NAMED_WIDGET TEXTFIELD named CH4VOLT and the BUTTON that reads the TEXTFIELD.
<PRE>
SCREEN AUTO AUTO 1.0

VERTICALBOX
  HORIZONTAL
    VERTICAL
      LABEL "Chan"
      LABEL "4"
    END
    VERTICAL
      LABEL "Volts"
      NAMED_WIDGET CH4VOLT TEXTFIELD 12 "5.0"
    END
  END
END

HORIZONTALBOX
  BUTTON 'Set' "api.cmd('POWER VOLTAGE_SET with CHANNEL 4, VOLTAGE '+screen.get_named_widget('CH4VOLT').text())"
END
</PRE>

When I open this in COSMOS (SHOW SCREEN), the TEXTFIELD is not displayed.  It should show "5.0".  This can be seen in the attached screen shot.  Pressing the BUTTON sends the command (with the default 5.0).
![screen1](https://user-images.githubusercontent.com/111520678/192549322-65f19f45-4c96-4efa-8816-1fc2937d9ee0.png)

I have discovered that if I click the screen's edit button (the pencil icon), then hit Cancel, the TEXTFIELD appears.  This can be seen in the second screen shot.
![screen2](https://user-images.githubusercontent.com/111520678/192549613-60a1a60a-5809-4e03-b756-98798740f093.png)

However, now pressing the BUTTON results in an error:
<PRE>
[
    {
        "type": "error",
        "message": "TypeError: screen.get_named_widget(...) is undefined",
        "time": 1664286550115
    }
]
</PRE>

I am using COSMOS 5.0.4 on Linux; the browser is Firefox.

**Comment by ryanmelt** [comment]
We've made several fixes to TlmViewer since 5.0.4.  Can you try our 5.0.10 release?

**Comment by jmthomas** [comment]
@jhayes-apl have you had a chance to try this in the latest release?

**Comment by jhayes-apl** [comment]
It is on my "to do" list.  I will let you know the results.

**Comment by jhayes-apl** [comment]
We have switched to OpenC3 5.0.11.  There were two problems originally (5.0.4):
1. The values in a TEXTFIELD do not appear on the screen
2. If you click edit (the pencil), the value would appear, but button code couldn't access the value.

Number 2 is fixed; number 1 persists.  I created three simple screens to illustrate the problem, power4.txt, power5.txt, and power6.txt.  These are shown in the following screen shot; only power5.txt shows the TEXTFIELD values:
![textfield](https://user-images.githubusercontent.com/111520678/196506146-1141d0c0-043d-4f7b-b0f1-0a352bd54dd7.png)
Here are the files, power4.txt:
<PRE>
SCREEN AUTO AUTO 1.0

MATRIXBYCOLUMNS 2
  LABEL "Chan"
  LABEL "V"
  LABEL "#3"
  NAMED_WIDGET CH3VOLT TEXTFIELD 12 "10.0"
  LABEL "#4"
  NAMED_WIDGET CH4VOLT TEXTFIELD 12 "12.0"
END

HORIZONTALBOX
  BUTTON 'Set 3' "api.cmd('POWER VOLTAGE_SET with CHANNEL 3, VOLTAGE '+screen.get_named_widget('CH3VOLT').text())"
  BUTTON 'Set 4' "api.cmd('POWER VOLTAGE_SET with CHANNEL 4, VOLTAGE '+screen.get_named_widget('CH4VOLT').text())"
END
</PRE>

And, power5.txt (the one that works):
<PRE>
SCREEN AUTO AUTO 1.0

VERTICAL
  HORIZONTAL
    LABEL "Ch"
    LABEL "V"
  END
  HORIZONTAL
    LABEL "#3"
    NAMED_WIDGET CH3VOLT TEXTFIELD 12 "10.0"
  END
  HORIZONTAL
    LABEL "#4"
    NAMED_WIDGET CH4VOLT TEXTFIELD 12 "12.0"
  END
END

HORIZONTAL
  BUTTON 'Set 3' "api.cmd('POWER VOLTAGE_SET with CHANNEL 3, VOLTAGE '+screen.get_named_widget('CH3VOLT').text())"
  BUTTON 'Set 4' "api.cmd('POWER VOLTAGE_SET with CHANNEL 4, VOLTAGE '+screen.get_named_widget('CH4VOLT').text())"
END
</PRE>

And, power6.txt:
<PRE>
SCREEN AUTO AUTO 1.0

HORIZONTAL
  VERTICAL
    LABEL "Ch"
    LABEL "#3"
    LABEL "#4"
  END
  VERTICAL
    LABEL "V"
    NAMED_WIDGET CH3VOLT TEXTFIELD 12 "10.0"
    NAMED_WIDGET CH4VOLT TEXTFIELD 12 "12.0"
  END
END

HORIZONTAL
  BUTTON 'Set 3' "api.cmd('POWER VOLTAGE_SET with CHANNEL 3, VOLTAGE '+screen.get_named_widget('CH3VOLT').text())"
  BUTTON 'Set 4' "api.cmd('POWER VOLTAGE_SET with CHANNEL 4, VOLTAGE '+screen.get_named_widget('CH4VOLT').text())"
END
</PRE>
